src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

```diff
@@ -86,4 +110,122 @@ public function update(OrderReturn $orderReturn): void
+            );
+        }
+        $orderReturnDetail = $this->getOrderReturnDetailByOrderDetailId($orderReturnDetailId->getValue());
+
+        $this->deleteOrderReturnDetail(
+            $orderReturnDetail->getOrderReturnId(),
+            $orderReturnDetailId,
+            $orderReturnDetail->getCustomizationId()
+        );
+    }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param OrderReturnDetailId $orderReturnDetailId
+     * @param CustomizationId|null $customizationId
+     *
+     * @throws DeleteOrderReturnProductException
+     */
+    public function deleteOrderReturnDetail(
+        OrderReturnId $orderReturnId,
+        OrderReturnDetailId $orderReturnDetailId,
+        ?CustomizationId $customizationId = null
+    ): void {
+        try {
+            if (!OrderReturn::deleteOrderReturnDetail(
+                $orderReturnId->getValue(),
+                $orderReturnDetailId->getValue(),
+                $customizationId ? $customizationId->getValue() : 0
+            )) {
+                throw new DeleteOrderReturnProductException(
+                    'Failed to delete merchandise return detail',
+                    DeleteOrderReturnProductException::UNEXPECTED_ERROR
+                );
+            }
+        } catch (PrestaShopException $e) {
+            throw new DeleteOrderReturnProductException(
+                'Failed to delete merchandise return detail',
+                DeleteOrderReturnProductException::UNEXPECTED_ERROR
+            );
+        }
+    }
+
+    /**
+     * @param int $orderDetailId
+     *
+     * @return OrderReturnDetail
+     *
+     * @throws OrderReturnDetailNotFoundException
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function getOrderReturnDetailByOrderDetailId(int $orderDetailId): OrderReturnDetail
+    {
+        $result = $this->connection->createQueryBuilder()
+            ->select('id_order_return, id_order_detail, id_customization, product_quantity')
+            ->from($this->dbPrefix . 'order_return_detail')
+            ->where('id_order_detail = :orderDetailId')
+            ->setParameter('orderDetailId', $orderDetailId)
+            ->execute()->fetchAssociative();
+        if (!$result) {
+            throw new OrderReturnDetailNotFoundException(sprintf('Order return detail with id "%s" not found', $orderDetailId));
+        }
+
+        return new OrderReturnDetail(
+            (int) $result['id_order_return'],
+            (int) $result['id_order_detail'],
+            (int) $result['product_quantity'],
+            (int) $result['id_customization'] ?: null
+        );
+    }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param Order $order
+     *
+     * @return array<OrderReturnDetail>
+     *
+     * @throws OrderReturnException
+     */
+    public function getOrderReturnDetails(OrderReturnId $orderReturnId, Order $order): array
```

Comment:
```suggestion
    public function getOrderReturnDetails(OrderReturnId $orderReturnId): array
```
Considering how the `$order` is used in this legacy method `OrderReturn::getOrdersReturnProducts` I agree it may be a bit too long to refacto for now. However since the `$order` is never used in the handler that calls this method I think the order fetching should be done here, instead of inject the `OrderRepository` in the handle you should inject it here and get the Order object at the beginning of this method (you can get the order ID via a DBAL query or by fetch the OrderReturn object as you prefer)

Reply:
Makes sense, refactored it